pkg/catalog/routes.go

```diff
@@ -435,3 +435,19 @@ func (mc *MeshCatalog) routesFromRules(rules []target.TrafficTargetRule, traffic
 
 	return routes, nil
 }
+
+// GetServicesForServiceAccounts returns a list of services corresponding to a list service accounts
+func (mc *MeshCatalog) GetServicesForServiceAccounts(saList []service.K8sServiceAccount) []service.MeshService {
```

Comment:
Why this function instead of using this one? https://github.com/openservicemesh/osm/blob/59a2722b478433c790b0cc58d79c54f65280c937/pkg/catalog/service.go#L10

Reply:
Created this mainly so I could loop through the list and log the error instead of returning an error. The err being returned is currently being used by other functions in the codebase and I didn't want to change current behavior. Once the routes refactor is complete, we could merge the two functions and use one although I kinda like the way it reads now.

Reply:
It will help to track future refactoring like this via a comment referencing an issue so that it doesn't get left behind.

Reply:
Okay that makes sense. I agree, if you add a comment, then this seems good to go

Reply:
all set.